vavr/src/test/java/org/jdbi/v3/vavr/TestVavrOptionMapperWithDB.java

```diff
@@ -0,0 +1,110 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.vavr;
+
+import io.vavr.collection.Set;
+import io.vavr.control.Option;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.generic.GenericType;
+import org.jdbi.v3.core.mapper.reflect.ConstructorMapper;
+import org.jdbi.v3.core.rule.H2DatabaseRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.sql.SQLException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestVavrOptionMapperWithDB {
+
+    @Rule
+    public H2DatabaseRule dbRule = new H2DatabaseRule().withPlugins();
+
+    @Before
+    public void addData() {
+        Handle handle = dbRule.openHandle();
+        handle.createUpdate("insert into something (id, name) values (1, 'eric')").execute();
+        handle.createUpdate("insert into something (id) values (2)").execute();
+    }
+
+    @Test
+    public void testOptionMapped_shouldSucceed() throws SQLException {
+        final Set<Option<String>> result = dbRule.getSharedHandle()
+                .registerRowMapper(ConstructorMapper.factory(SomethingWithOption.class))
+                .createQuery("select name from something")
+                .collectInto(new GenericType<Set<Option<String>>>() {});
+
+        assertThat(result).hasSize(2);
+        assertThat(result).contains(Option.none(), Option.of("eric"));
+    }
+
+    @Test
+    public void testOptionMappedWithinObjectIfPresent_shouldContainValue() throws SQLException {
+        final SomethingWithOption result = dbRule.getSharedHandle()
+                .registerRowMapper(ConstructorMapper.factory(SomethingWithOption.class))
+                .createQuery("select id, name from something where id = 1")
+                .mapTo(SomethingWithOption.class)
+                .findOnly();
+
+        assertThat(result.getName()).isInstanceOf(Option.class);
+        assertThat(result.getName().get()).isEqualToIgnoringCase("eric");
+    }
+
+    @Test
+    public void testOptionWithinObjectIfMissing_shouldBeNone() throws SQLException {
+        final SomethingWithOption result = dbRule.getSharedHandle()
+                .registerRowMapper(ConstructorMapper.factory(SomethingWithOption.class))
+                .createQuery("select id, name from something where id = 2")
+                .mapTo(SomethingWithOption.class)
+                .findOnly();
+
+        assertThat(result.getName()).isInstanceOf(Option.class);
+        assertThat(result.getName().getOrNull()).isNull();
```

Comment:
Wouldn't hurt to add `assertj-vavr` as a test dependency to `jdbi3-vavr` so you can do fluent assertions on vavr types.

https://github.com/assertj/assertj-vavr

Reply:
There is no stable release yet and i cannot find it in any repository.

Reply:
:disappointed:

Reply:
https://github.com/assertj/assertj-vavr/issues/8

Reply:
@alwins0n is right - it's not done yet. If you're looking for a workaround, you can simply convert Vavr's `Option` to Java 8's `Optional` and leverage dedicated assertions

Reply:
I already rewrote the test suite for this pull request and decided to equal on the result object `SomethingWithOption` as a whole which makes a lot more sense (since it can verify also the correct record via id)